Reachability.m

```diff
@@ -152,6 +156,8 @@ -(instancetype)initWithReachabilityRef:(SCNetworkReachabilityRef)ref
         self.reachableOnWWAN = YES;
         self.reachabilityRef = ref;
 
+        CFRetain(self.reachabilityRef);
```

Comment:
Right, so my implementation fixes that. Unless I'm missing something.

Line 98, `SCNetworkReachabilityRef` is created, retain count 1.
Line 101, init does something with it. Sucess or fail, doesn't matter. As far as we know, the retain count is still 1. So ...
Line 103, we release it. If other objects are claiming ownership its one them